rfcs/2022-12-09-541-reload-api.md

```diff
@@ -0,0 +1,208 @@
+# RFC 541 - 2022-12-09 - Reload API
+
+This RFC proposes adding an HTTP API for Vector's configuration reloading
+process. This will allow an enterprise control plane to more directly control a
+running instance and provide a path for synchronous runtime feedback that
+doesn't currently exist.
+
+## Context
+
+- [RFC: Observability Pipelines Worker (internal)](https://docs.google.com/document/d/10c6KuzbUcEUJxBfzX3a2CQaLHGPW2zKlz-_U8kS2DmA)
+- [Observability Pipelines Worker Implementation (internal)](https://docs.google.com/document/d/1hXv0fO1nsMkZeBIgZAsGD6QB9NACYaioxQ1FoQz94LI)
+
+## Cross cutting concerns
```

Comment:
Another cross-cutting concern might be reloading in Kubernetes. I realize that could be out-of-scope for this RFC, specifically, but is something we need to solve as well and could influence the design here.

Reply:
That's one I'm not terribly familiar with. If anyone could weigh in with some description of the problem there I'd be happy to address it.

Reply:
Yeah, sorry, I was thinking about updating the container resources like forwarded ports which is outside of Vector's configuration, but still something we'll need to solve for a remotely configured Vector.

Reply:
Correct me if I'm wrong, but that seems more like something that would be addressed in a k8s operator than a Vector API? I'm not sure what I could do here to make that better.

Reply:
Indeed, those would have to happen at a higher abstraction layer - and ironically likely end up not using this API at all, as the resources would likely need to be recreated.